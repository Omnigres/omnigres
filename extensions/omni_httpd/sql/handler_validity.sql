-- Invalid queries
BEGIN;
INSERT INTO omni_httpd.handlers VALUES (DEFAULT) RETURNING id AS handler_id \gset
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$SELECT * FROM no_such_table$$);
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$SELECT request.pth FROM request$$);
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$$$);
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$SELECT; SELECT$$);
ROLLBACK;

-- Valid query at the end of the transaction
BEGIN;
INSERT INTO omni_httpd.handlers VALUES (DEFAULT) RETURNING id AS handler_id \gset
INSERT INTO omni_httpd.handlers_queries (handler_id, query) VALUES(:handler_id, $$SELECT * FROM no_such_table$$);
CREATE TABLE no_such_table ();
END;

DROP TABLE no_such_table;
