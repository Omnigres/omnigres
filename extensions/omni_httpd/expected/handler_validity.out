-- Invalid queries
BEGIN;
INSERT INTO omni_httpd.handlers VALUES (DEFAULT) RETURNING id AS handler_id \gset
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$SELECT * FROM no_such_table$$);
ERROR:  relation "omni_httpd.handler_queries" does not exist
LINE 1: INSERT INTO omni_httpd.handler_queries (handler_id, query) V...
                    ^
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$SELECT request.pth FROM request$$);
ERROR:  current transaction is aborted, commands ignored until end of transaction block
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$$$);
ERROR:  current transaction is aborted, commands ignored until end of transaction block
INSERT INTO omni_httpd.handler_queries (handler_id, query) VALUES(:handler_id, $$SELECT; SELECT$$);
ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK;
-- Valid query at the end of the transaction
BEGIN;
INSERT INTO omni_httpd.handlers VALUES (DEFAULT) RETURNING id AS handler_id \gset
INSERT INTO omni_httpd.handlers_queries (handler_id, query) VALUES(:handler_id, $$SELECT * FROM no_such_table$$);
CREATE TABLE no_such_table ();
END;
DROP TABLE no_such_table;
